src/huggingface_hub/cli/repo_files.py

```diff
@@ -34,95 +34,89 @@
+
+
+@repo_files_cli.command("delete")
+def repo_files_delete(
+    repo_id: Annotated[
+        str,
+        typer.Argument(
+            ...,
+            help="The ID of the repo (e.g. username/repo-name).",
+        ),
+    ],
+    patterns: Annotated[
+        list[str],
+        typer.Argument(
+            ...,
             help="Glob patterns to match files to delete.",
-        )
-        delete_subparser.add_argument(
-            "--repo-type",
-            choices=["model", "dataset", "space"],
-            default="model",
+        ),
+    ],
+    repo_type: Annotated[
+        Optional[str],
+        typer.Option(
             help="Type of the repo to upload to (e.g. `dataset`).",
-        )
-        delete_subparser.add_argument(
-            "--revision",
-            type=str,
-            help=(
-                "An optional Git revision to push to. It can be a branch name "
-                "or a PR reference. If revision does not"
-                " exist and `--create-pr` is not set, a branch will be automatically created."
-            ),
-        )
-        delete_subparser.add_argument(
-            "--commit-message", type=str, help="The summary / title / first line of the generated commit."
-        )
-        delete_subparser.add_argument(
-            "--commit-description", type=str, help="The description of the generated commit."
-        )
-        delete_subparser.add_argument(
-            "--create-pr", action="store_true", help="Whether to create a new Pull Request for these changes."
-        )
-        delete_subparser.add_argument(
+        ),
+    ] = "model",
+    revision: Annotated[
+        Optional[str],
+        typer.Option(
+            help="An optional Git revision to push to. It can be a branch name or a PR reference. If revision does not exist and `--create-pr` is not set, a branch will be automatically created.",
+        ),
+    ] = None,
+    commit_message: Annotated[
+        Optional[str],
+        typer.Option(
+            help="The summary / title / first line of the generated commit.",
+        ),
+    ] = None,
+    commit_description: Annotated[
+        Optional[str],
+        typer.Option(
+            help="The description of the generated commit.",
+        ),
+    ] = None,
+    create_pr: Annotated[
+        bool,
+        typer.Option(
+            "--create-pr",
+            help="Whether to create a new Pull Request for these changes.",
+        ),
+    ] = False,
+    token: Annotated[
+        Optional[str],
+        typer.Option(
             "--token",
-            type=str,
             help="A User Access Token generated from https://huggingface.co/settings/tokens",
-        )
-
-        repo_files_parser.set_defaults(func=RepoFilesCommand)
+        ),
+    ] = None,
+) -> None:
+    logging.set_verbosity_info()
+    repo_type = validate_repo_type(repo_type)
+    api = HfApi(token=token, library_name="hf")
```

Comment:
That's actually a really good idea to track usage of our CLI. I think it'd be good to centralize this (some other `HfApi` instantiations are not defining the library_name) but can be done in a later PR. (we have a subticket about analytics specifically)

Reply:
```suggestion
    api = HfApi(token=token, library_name="hf", library_version=__version__)
```

might be nice to add `huggingface_hub` version as well (would make things easier to works with in Kibana even though we already have the info in `hf_hub` user agent)